src/sentry/ingest/billing_metrics_consumer.py

```diff
@@ -0,0 +1,156 @@
+from arroyo.types import Message, Partition, Position
+from django.conf import settings
+
+from sentry.constants import DataCategory
+from sentry.sentry_metrics.indexer.strings import TRANSACTION_METRICS_NAMES
+from sentry.utils import json
+from sentry.utils.outcomes import Outcome, track_outcome
+
+logger = logging.getLogger(__name__)
+
+
+def get_metrics_billing_consumer(
+    topic: str,
+    group_id: str,
+    auto_offset_reset: str,
+    force_topic: Union[str, None],
+    force_cluster: Union[str, None],
+    **options: Any,
+) -> StreamProcessor[KafkaPayload]:
+    bootstrap_servers = _get_bootstrap_servers(topic, force_topic, force_cluster)
+
+    logger.warning("Unused options: %s", options)
+
+    return StreamProcessor(
+        consumer=KafkaConsumer(
+            build_kafka_consumer_configuration(
+                default_config={},
+                group_id=group_id,
+                auto_offset_reset=auto_offset_reset,
+                bootstrap_servers=bootstrap_servers,
+            ),
+        ),
+        topic=Topic(topic),
+        processor_factory=BillingMetricsConsumerStrategyFactory(),
+    )
+
+
+def _get_bootstrap_servers(
+    kafka_topic: str, force_topic: Union[str, None], force_cluster: Union[str, None]
+) -> Sequence[str]:
+    topic = force_topic or kafka_topic
+    cluster = force_cluster or settings.KAFKA_TOPICS[topic]["cluster"]
+
+    options = settings.KAFKA_CLUSTERS[cluster]
+    servers = options["common"]["bootstrap.servers"]
+    if isinstance(servers, (list, tuple)):
+        return servers
+    return [servers]
+
+
+class BillingMetricsConsumerStrategyFactory(ProcessingStrategyFactory[KafkaPayload]):
+    def create_with_partitions(
+        self,
+        commit: Callable[[Mapping[Partition, Position]], None],
+        partitions: Mapping[Partition, int],
+    ) -> ProcessingStrategy[KafkaPayload]:
+        return BillingTxCountMetricConsumerStrategy(commit)
+
+
+class MetricsBucket(TypedDict):
+    """
+    Metrics bucket as decoded from kafka.
+
+    Only defines the fields that are relevant for this consumer."""
+
+    org_id: int
+    project_id: int
+    metric_id: int
+    timestamp: int
+    value: Sequence[float]  # only true for distribution buckets
+
+
+class BillingTxCountMetricConsumerStrategy(ProcessingStrategy[KafkaPayload]):
+    """A metrics consumer that generates a billing outcome for each processed
+    transaction, processing a bucket at a time. The transaction count is
+    computed from the amount of values from `d:transactions/duration@millisecond`
+    buckets.
+    """
+
+    counter_metric_id = TRANSACTION_METRICS_NAMES["d:transactions/duration@millisecond"]
+
+    def __init__(self, commit: Callable[[Mapping[Partition, Position]], None]) -> None:
+        self.__commit = commit
+        self.__ready_to_commit: MutableMapping[Partition, Position] = {}
+        self.__closed = False
+
+    def poll(self) -> None:
+        self._bulk_commit()
```

Comment:
@lynnagara is it OK to commit on every poll, or do we need batched or throttled commits, like in 

https://github.com/getsentry/sentry/blob/6da3b229f74e1fe82b30d775c112e0490a41f22b/src/sentry/replays/consumers/recording/process_recording.py#L249-L256

Reply:
See https://github.com/getsentry/sentry/blob/f94d5af03ed666db8b5ad3898641bdb785ee17e4/src/sentry/runner/commands/run.py#L452-L466 to decide how often to commit.